packages/babel-plugin-transform-typescript/test/fixtures/exports/declare-shadowed/input.ts

```diff
@@ -0,0 +1,9 @@
+declare class Signal<T = any> {
+	value: T
+}
+
+function Signal(this: Signal, value?: unknown) {
+	this.value = value
+}
+
+export { Signal };
```

Comment:
Can also add a test with the declarations swapped?

```ts
declare class Signal<T = any> {
	value: T
}

function Signal(this: Signal, value?: unknown) {
	this.value = value
}

export { Signal };
```

Reply:
ts seems to show that it cannot be merged.